pallets/runtime/tests/src/contract_test.rs

```diff
@@ -154,17 +158,35 @@ where
     T::Hashing::hash(&b"abc".encode())
 }
 
+/// Executes `f` on a created `TestExternalities` using the given `network_fee_share` and `protocol_base_fees`.
+///
+/// It also enables the `put_code` extrinsics.
+fn execute_externalities_with_wasm<T, F>(
+    network_fee_share: Perbill,
+    protocol_base_fees: MockProtocolBaseFees,
+    f: F,
+) where
+    F: FnOnce(Vec<u8>, <T::Hashing as Hash>::Output) -> (),
+    T: frame_system::Trait,
+{
+    let (wasm, code_hash) = compile_module::<T>("flipper").unwrap();
+    ExtBuilder::default()
+        .network_fee_share(network_fee_share)
```

Comment:
```suggestion
fn execute_externalities_with_wasm(
    network_fee_share: u32,
    protocol_base_fees: MockProtocolBaseFees,
    test: impl FnOnce(Vec<u8>, <TestStorage::Hashing as Hash>::Output),
) {
    let (wasm, code_hash) = compile_module::<TestStorage>("flipper").unwrap();
    ExtBuilder::default()
        .network_fee_share(Perbill::from_percent(network_fee_share))
```
This is test code, so `T == TestStorage` should always be the case and you're always using `from_percent`.

Reply:
I would like to keep the generalized version because there is not requirement to be explicitly a `TestStorage`.
About `network_fee_share`, I'd also rather to use `Perbill` type than a plain `u32`.

Reply:
When it is going to be something else than `T == TestStorage`? Generalization should add some value, and if it doesn't, you're just making all call-sites and the definition harder to read. Same with `Perbill`, that also increases duplication, adding more code to read.

Reply:
As this is not the main target of this task, I am going to create a external refactor task to handle this.